Resources/Management/map_scripts.gml

```diff
@@ -0,0 +1,16 @@
+/// map_scripts(map)
+//
+//  Fills a given map data structure with the id values of all
+//  script resources, keyed by strings of their names.
+//
+//      map         map data structure, real
+//
+/// GMLscripts.com/license
+{
+    var i, ds_map;
+    ds_map = argument0;
+    for (i = 0; script_exists(i); i += 1) {
```

Comment:
Last time I've checked, the resource ID's are not necessarily contiguous. For example if you create three scripts, then delete the middle one, the id's will be `[0, 2]`. Has this changed since?

Reply:
No it's still like that.

Reply:
Could be worth a check, maybe something like "haven't seen an existing script in the last 1024 iterations, let's just call it a day and return".